src/App.tsx

```diff
@@ -1,26 +1,195 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState } from 'react';
+import { Todo } from './types/Todo';
+import cn from 'classnames';
+import { TodoItem } from './components/TodoItem';
+import { ErrorNotification } from './components/ErrorsNotification';
+import { useTodos } from './hooks/useTodos';
 
-const USER_ID = 0;
+type Sort = 'All' | 'Active' | 'Completed';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const {
+    todos,
+    title,
+    setTitle,
+    tempState,
+    loadingIds,
+    handleSubmit,
+    handleDelete,
+    handleUpdate,
+    inputRef,
+  } = useTodos();
+  const [sortBy, setSortBy] = useState<Sort>('All');
+
+  const activeTodo = [...todos].filter(todo => todo.completed === false);
+  const completedTodo = [...todos].filter(todo => todo.completed === true);
+
+  let filteredTodos = todos;
+
+  if (sortBy === 'Active') {
+    filteredTodos = activeTodo;
   }
 
+  if (sortBy === 'Completed') {
+    filteredTodos = completedTodo;
+  }
+  //   const completedTodo: Todo[] = todos.map(todo => {
+  //     if (todo.id === id) {
+  //       setLoadingIds([...loadingIds, todo.id]);
```

Comment:
Few issues here

1. Please move `type Sort` and sorting logic away from the component
2. Don't keep commented code
3. You are filtering in a bad way - your code will perform filtering (ln 24-25) in any case, regardless if it is needed or not. Please review your logic.

Reply:
Hi! I think, we need filtering always apart from the situation when we don`t have any todos